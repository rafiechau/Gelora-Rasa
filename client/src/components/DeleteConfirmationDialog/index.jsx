import { Button, Dialog, DialogActions, DialogContent, DialogContentText, DialogTitle } from '@mui/material';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

const DeleteConfirmationDialog = ({ open, onConfirm, onCancel, dialogTitle, dialogContent }) => (
  <Dialog
    open={open}
    onClose={onCancel}
    aria-labelledby="alert-dialog-title"
    aria-describedby="alert-dialog-description"
    data-testid="delete-confirmation-dialog"
  >
    <DialogTitle id="alert-dialog-title">{dialogTitle}</DialogTitle>
    <DialogContent>
      <DialogContentText id="alert-dialog-description">{dialogContent}</DialogContentText>
    </DialogContent>
    <DialogActions>
      <Button onClick={onCancel} data-testid="cancel-button">
        <FormattedMessage id="app_cancel_dialog" />
      </Button>
      <Button onClick={onConfirm} autoFocus data-testid="confirm-button">
        <FormattedMessage id="app_delete_dialog" />
      </Button>
    </DialogActions>
  </Dialog>
);

DeleteConfirmationDialog.propTypes = {
  open: PropTypes.bool.isRequired,
  onConfirm: PropTypes.func.isRequired,
  onCancel: PropTypes.func.isRequired,
  dialogTitle: PropTypes.node.isRequired,
  dialogContent: PropTypes.node.isRequired,
};

export default DeleteConfirmationDialog;
