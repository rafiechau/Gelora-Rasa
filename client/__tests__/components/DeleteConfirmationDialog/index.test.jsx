import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import DeleteConfirmationDialog from '@components/DeleteConfirmationDialog';

jest.mock('react-intl', () => ({
  FormattedMessage: jest.fn(({ id }) => <span>{id}</span>),
}));

describe('DeleteConfirmationDialog', () => {
  const mockOnConfirm = jest.fn();
  const mockOnCancel = jest.fn();

  beforeEach(() => {
    render(
      <DeleteConfirmationDialog
        open
        onConfirm={mockOnConfirm}
        onCancel={mockOnCancel}
        dialogTitle="Confirm Delete"
        dialogContent="Are you sure you want to delete this?"
      />
    );
  });

  test('Should render correctly', () => {
    expect(screen.getByTestId('delete-confirmation-dialog')).toBeInTheDocument();
  });

  test('Should call onCancel when cancel button is clicked', () => {
    const cancelButton = screen.getByTestId('cancel-button');
    fireEvent.click(cancelButton);
    expect(mockOnCancel).toHaveBeenCalled();
  });

  test('Should call onConfirm when confirm button is clicked', () => {
    const confirmButton = screen.getByTestId('confirm-button');
    fireEvent.click(confirmButton);
    expect(mockOnConfirm).toHaveBeenCalled();
  });

  test('Should match snapshot', () => {
    const dialog = screen.getByTestId('delete-confirmation-dialog');
    expect(dialog).toMatchSnapshot();
  });
});
