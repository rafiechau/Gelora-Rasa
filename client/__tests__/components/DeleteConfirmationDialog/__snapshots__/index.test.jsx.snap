// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteConfirmationDialog Should match snapshot 1`] = `
<div
  class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
  data-testid="delete-confirmation-dialog"
  role="presentation"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    data-testid="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
    role="presentation"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    tabindex="-1"
  >
    <div
      aria-describedby="alert-dialog-description"
      aria-labelledby="alert-dialog-title"
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
      role="dialog"
    >
      <h2
        class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
        id="alert-dialog-title"
      >
        Confirm Delete
      </h2>
      <div
        class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
      >
        <p
          class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiDialogContentText-root css-qfso29-MuiTypography-root-MuiDialogContentText-root"
          id="alert-dialog-description"
        >
          Are you sure you want to delete this?
        </p>
      </div>
      <div
        class="MuiDialogActions-root MuiDialogActions-spacing css-hlj6pa-MuiDialogActions-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          data-testid="cancel-button"
          tabindex="0"
          type="button"
        >
          <span>
            app_cancel_dialog
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium Mui-focusVisible MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          data-testid="confirm-button"
          tabindex="0"
          type="button"
        >
          <span>
            app_delete_dialog
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          >
            <span
              class="css-y4cjyz-MuiTouchRipple-ripple MuiTouchRipple-ripple MuiTouchRipple-rippleVisible MuiTouchRipple-ripplePulsate"
              style="width: 1px; height: 1px; top: -0.5px; left: -0.5px;"
            >
              <span
                class="MuiTouchRipple-child MuiTouchRipple-childPulsate"
              />
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    data-testid="sentinelEnd"
    tabindex="0"
  />
</div>
`;
